### ISSUE TITLE
Network Error if target is localhost

### ISSUE THREAD

**Issue by distahl** [alessandromaggio/pythonping#15]
Hi,
I'm getting this if I try to ping localhost or any other ip which is local:

```
Python 3.6.2 (default, Aug 17 2017, 09:31:39) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-17)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pythonping import ping, executor
>>> ping('127.0.0.1', verbose=True)
Network Error from 127.0.0.1 in 0.02ms
Reply from 127.0.0.1, 9 bytes in 0.01ms
Network Error from 127.0.0.1 in 0.01ms
Reply from 127.0.0.1, 9 bytes in 0.01ms
Network Error from 127.0.0.1 in 0.02ms
Reply from 127.0.0.1, 9 bytes in 0.01ms
Network Error from 127.0.0.1 in 0.01ms
Reply from 127.0.0.1, 9 bytes in 0.01ms
```

It alternates always between "Network Error" and "Reply"
Any idea?

EDIT:
Looks like every time this "Network Error" happens, the message type is 8, which means "ECHO REQUEST". So I think we see our own packages we send before via ping coming back in.

EDIT2:

This seems to resolve the error for me:
``` patch
--- pythonping/executor.py       2019-06-14 07:49:30.000000000 +0200
+++ pythonping/executor.py   2019-06-14 08:31:44.374020000 +0200
@@ -233,7 +233,7 @@
             # If we actually received something
             if raw_packet != b'':
                 response.unpack(raw_packet)
-                if response.id == packet_id:
+                if response.id == packet_id and response.message_type != 8:
                     return Response(Message('', response, source_socket[0]), timeout-time_left)
         return Response(None, timeout)
 ```

**Comment by alessandromaggio** [comment]
You made a pretty good troubleshooting. Still, I've never seen this issue before, I would like to replicate it. What OS and version are you running?
If the root cause is the one you provided, and this snippet is effectively a solution and not a workaround, I will release a new version to fix it.

**Comment by distahl** [comment]
Hi,
It was RHEL6 (if I remember right, RHEL 7 was also affected) with Python 3.6.2. If you don't have access to RHEL, you should be able to use CentOS, which should be the same.

If you can't reproduce this, please try a second thing which maybe caused this. Ping the system which is executing the python script from somewhere else, while executing the script.

**Comment by alessandromaggio** [comment]
Thank you, I will look into that ASAP and release an hotfix.

**Comment by alessandromaggio** [comment]
Integrated with PR #18, bug is fixed in version 1.0.4 already released (also on PyPI).
Thanks!
